rita_client/src/operator_update/mod.rs

```diff
@@ -335,6 +350,60 @@ fn perform_operator_update(
     trace!("Successfully completed OperatorUpdate");
 }
 
+fn update_authorized_keys(
+    add_list: Vec<String>,
+    drop_list: Vec<String>,
+    keys_file: &str,
+) -> Result<(), std::io::Error> {
+    let mut existing_keys = Vec::new();
+    let mut keys_to_add = Vec::new();
+    let now = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .unwrap()
+        .as_secs();
+
+    let managed_by = String::from("//managed by OpsTools");
+
+    let key_file = File::open(keys_file);
+    match key_file {
+        Ok(key_file_open) => {
+            let buf_reader = BufReader::new(key_file_open);
+
+            for key in buf_reader.lines() {
+                print!("leaving existing key");
+                existing_keys.push(key.unwrap());
+            }
+            if !add_list.is_empty() {
```

Comment:
this if statement isn't needed the for loop will simply do nothing if the list is empty.